### PR TITLE
Discord bot token and Grafana service

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- Add Discord bot token pattern and filter ([test sample](https://github.com/discord/sample-game-integration/blob/master/server/example.cfg))
- Add Grafana service token pattern and filter (tested for tokens from [opensource repos](https://github.com/grafana/grafana/blob/main/public/openapi3.json))
- skip .map  .rc   .vdproj   .rc2   files from scan
- actualize Facebook token pattern
- fix quotation keyword pattern (skip some filters for well quoted values)

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest
- [x] Benchmark

### CODE REVIEW

**Review by xDizzix — APPROVED** [review]
+

File: credsweeper/common/constants.py [hunk 1]
```diff
@@ -17,8 +17,10 @@ class KeywordPattern:
     # Authentication scheme ( oauth | basic | bearer | apikey ) precedes to credential
     value = r"(?P<value_leftquote>((b|r|br|rb|u|f|rf|fr|\\{0,8})?[`'\"]){1,4})?" \
             r"( ?(oauth|bot|basic|bearer|apikey|accesskey) )?" \
-            r"(?P<value>(?:\{[^}]{3,8000}\})|(?:<[^>]{3,8000}>)|" \
-            r"(?(value_leftquote)(?:\\[tnrux0-7][0-9a-f]*|[^`'\"\\])|(?:\\n|\\r|\\?[^\s`'\"\\,;])){3,8000})" \
+            r"(?P<value>" \
+            r"(?(value_leftquote)(?:\\[tnrux0-7][0-9a-f]*|[^`'\"\\])|(?:\\n|\\r|\\?[^\s`'\"\\,;])){3,8000}" \
+            r"|(?:\{[^}]{3,8000}\})|(?:<[^>]{3,8000}>)" \
+            r")" \
```
Comment (csh519) [inline credsweeper/common/constants.py:23]: Could you explain why those lines are switched?

File: credsweeper/rules/config.yaml [hunk 2]
```diff
@@ -644,7 +659,7 @@
   confidence: moderate
   type: pattern
   values:
-    - (?<![0-9A-Za-z_+-])(?P<value>SK[0-9a-fA-F]{32})(?![=0-9A-Za-z_+-])
+    - (?<![0-9A-Za-z_+-])(?P<value>SK[0-9A-Fa-f]{32})(?![=0-9A-Za-z_+-])
```
Comment (csh519) [inline credsweeper/rules/config.yaml:662]: Do they have difference? Did you changed it just for beautify?

File: .github/workflows/benchmark.yml [hunk 3]
```diff
@@ -160,14 +160,14 @@ jobs:
...
-    # put the benchmark in single job to keep constant environment during test
+    # put the benchmark in single job to keep constant environment during test python 3.8 is not applicable
     needs: [ download_data ]
 
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        python-version: [ "3.10", "3.9", "3.10", "3.11" ]
+        python-version: [ "3.9", "3.10", "3.11" ]
```
Comment (csh519) [inline .github/workflows/benchmark.yml:170]: If the benchmark is not available on python 3.8, then should we consider to set minimum version of CredSweeper as 3.9?

Comment (babenek) [reply .github/workflows/benchmark.yml:170]: There only speed benchmark limitation. CredSweeper can work wit 3.8.
As i see, https://pypistats.org/packages/credsweeper - the version is most popular. 
With any limitation of a package we will set the minimum to 3.9

Comment (csh519) [reply .github/workflows/benchmark.yml:170]: Okay. 3.8 is the most popular version...

File: credsweeper/common/constants.py (hunk 1 above)
Comment (babenek) [reply credsweeper/common/constants.py:23]: that regex collects full quoted value first "``[2ns8BH]ds$%^``"
previous - collects only in brackets "``[2ns8BH]``ds$%^"

File: credsweeper/rules/config.yaml (hunk 2 above)
Comment (babenek) [reply credsweeper/rules/config.yaml:662]: Yes, i try to unify the pattern with all rules to quick search them in future
The order relates ASCII table.

Comment (csh519) [reply credsweeper/rules/config.yaml:662]: Good idea.

File: credsweeper/common/constants.py (hunk 1 above)
Comment (csh519) [reply credsweeper/common/constants.py:23]: Hmm... why do you think bracket included value is correct value?
I think previous regex was intended to filter the bracket included values.

**Review by kmnls — APPROVED** [review]
Agree

**Review by Yullia — APPROVED** [review]
Approved

File: credsweeper/common/constants.py (hunk 1 above)
Comment (babenek) [reply credsweeper/common/constants.py:23]: https://github.com/Samsung/CredSweeper/actions/runs/9839488594/job/27161670040
- the test with random generation

Comment (csh519) [reply credsweeper/common/constants.py:23]: Okay i see.

**Pull request merged**